python/pyarrow/lib_gpu.pxd

```diff
@@ -0,0 +1,67 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Perhaps just a style issue, but it doesn't seem to make much sense to define a `pxd` file if we're using it from a single `pyx` file.

Reply:
It depends if we expect people to write `from pyarrow.foo cimport symbol`. I would argue for calling this `cuda.pxd`, with `cuda.pyx` (or `_cuda.pyx` plus `cuda.py`) so that Cython can import using `from pyarrow.cuda cimport ...` if desired